### PR TITLE
Jib log refactor

### PR DESCRIPTION
**Pull request by dev-gaur**

### CODE REVIEW

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java [hunk 1]
```diff
@@ -162,18 +170,24 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
...
             Jib.from(baseImage).containerize(Containerizer.to(targetImage)
-                    .addEventHandler(LogEvent.class, log(logger))
-                    .addEventHandler(TimerEvent.class, new TimerEventHandler(logger::debug))
-                    .addEventHandler(ProgressEvent.class, new ProgressEventHandler(logUpdate(logger))));
+                    .addEventHandler(LogEvent.class, JibServiceUtil::log)
+                    .addEventHandler(TimerEvent.class, new TimerEventHandler(message -> consoleLogger.log(LogEvent.Level.DEBUG, message)))
+                    .addEventHandler(ProgressEvent.class, new ProgressEventHandler(update -> consoleLogger
+                            .setFooter(ProgressDisplayGenerator.generateProgressDisplay(update.getProgress(), update.getUnfinishedLeafTasks())))));
+            singleThreadedExecutor.shutDownAndAwaitTermination(Duration.ofSeconds(timeOut));
         } catch (RegistryException | CacheDirectoryCreationException | InvalidImageReferenceException | IOException | ExecutionException | InterruptedException e) {
```
Comment (rohanKanojia) [inline core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: bdw, if you're shutting down your executor both in try and catch block; shouldn't you be doing it in finally?

File: plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/BuildMojo.java [hunk 2]
```diff
@@ -69,6 +69,8 @@
     @Parameter(property = "fabric8.build.jib", defaultValue = "false")
     private boolean isJib;
 
+    @Parameter(property = "fabric8.build.jib.timeOut", defaultValue = "60")
```
Comment (rohanKanojia) [inline plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/BuildMojo.java]: Please add some documentation about this flag or all jib related flags also.

**Comment by dev-gaur** [comment]
image


@rohanKanojia 

ConsoleLogger from Jib is asynchronous and event based so it required some synchronization.
ConsoleLogger was added here https://github.com/fabric8io/fabric8-maven-plugin/commit/cef5ab136af5d3b10177db0dcd06c2cd0ca4f44f .

The logs get scattered and disorganized whenever the build fails, if not synchronized with a singleThreadExecutor. The PR does that.

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java (hunk 1 above)
Comment (dev-gaur) [reply core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: **Where is my mind by Pixies plays...**

Sorry that skipped my mind.

**Comment by manusa** [comment]
Please, consider trying the suggestion we discussed yesterday first.

We're bringing back an ugly dependency that is inconsistent with the rest of our codebase (`com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger`) and a nasty and non-clear way to wait for `singleThreadedExecutor` thread to complete.

The only reason why `ConsoleLogger` prints messages before the maven build complete (failed/success) message is print is because the `singleThreadExecutor` in `ConsoleLogger` acts as a blocking queue for log messages posted there. So it will block execution until all messages sent to the logger are posted **before** the `shutDownAndAwaitTermination` method in `singleThreadExecutor` is invoked. This **doesn't** translate in your method waiting for jib build to finish which is what you're assuming here.

The only way to wait for jibContainerBuilder to finish its' parallel execution is to provide our own executor `setExecutorService(jibBuildExecutor)` and await termination of the submitted jobs. (`JibContainerBuilder` internally performs a `Future#get` of the jobs posted to the provided `ExecutorService`)

e.g. for buildContainer method: 
```
    static void buildContainer(JibContainerBuilder jibContainerBuilder, TarImage image, Logger logger)
    throws InterruptedException {
        final ExecutorService jibBuildExecutor = Executors.newSingleThreadExecutor();
        try {
            jibContainerBuilder.setCreationTime(Instant.now());
            jibContainerBuilder.containerize(Containerizer.to(image)
              .setExecutorService(jibBuildExecutor)
              .addEventHandler(LogEvent.class, log(logger))
              .addEventHandler(TimerEvent.class, new TimerEventHandler(logger::debug))
              .addEventHandler(ProgressEvent.class, new ProgressEventHandler(logUpdate(logger))));
        } catch (CacheDirectoryCreationException | IOException | ExecutionException | RegistryException ex) {
            logger.error("Unable to build the image tarball: ", ex);
            throw new IllegalStateException(ex);
        } catch (InterruptedException ex) {
            logger.error("Thread interrupted", ex);
            throw ex;
        }
        jibBuildExecutor.awaitTermination(THREAD_EXECUTOR_TIMEOUT_SECONDS, TimeUnit.SECONDS);
    }
```

File: doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc [hunk 3]
```diff
@@ -198,9 +198,13 @@ By default S2I is used.
 | `fabric8.build.strategy`
 
 | *isJib*
-a| If the effective <<build-mode,mode>> is `kubernetes` then this option sets the build mode to JIB to perform a dockerless JIB build.
+| If the effective <<build-mode,mode>> is `kubernetes` then this option sets the build mode to JIB to perform a dockerless JIB build.
 | `fabric8.build.jib`
 
+| *jibTimeOut*
```
Comment (dev-gaur) [inline doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc]: documentation added

Comment (dev-gaur) [reply doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc]: @rohanKanojia

**Comment by dev-gaur** [comment]
Closing this as #1798 takes care of the issue.

**Pull request closed without merging**